cf-testd/cf-testd.c

```diff
@@ -607,16 +620,153 @@ int main(int argc, char *argv[])
         Log(LOG_LEVEL_DEBUG, "Got report file contents: %s", config->report);
     }
 
-    Log(LOG_LEVEL_INFO, "Starting server...");
+    Log(LOG_LEVEL_INFO, "Starting server at %s...", config->address);
     fflush(stdout); // for debugging startup
 
-    int r = CFTestD_StartServer(config);
+    config->ret = CFTestD_StartServer(config);
+
     CFTestD_ConfigDestroy(config);
 
     /* we don't really need to do this here because the process is about the
      * terminate, but it's a good way the cleanup actually works and doesn't
      * cause a segfault or something */
     ServerTLSDeInitialize(&(config->priv_key), &(config->pub_key), &(config->ssl_ctx));
 
-    return r;
+    return NULL;
+}
+
+static void HandleSignal(int signum)
+{
+    switch (signum)
+    {
+    case SIGTERM:
+    case SIGINT:
+        // flush all logging before process ends.
+        fflush(stdout);
+        fprintf(stderr, "Terminating...\n");
+        TERMINATE = true;
+        break;
+    default:
+        break;
+    }
+}
+
+/**
+ * @param ip_str string representation of an IPv4 address (the usual one, with
+ *               4 numbers separated by dots)
+ */
+static char *IncrementIPaddress(const char *ip_str)
+{
+    struct in_addr ip;
+    int ret = inet_aton(ip_str, &ip);
+    if (ret == 0)
+    {
+        Log(LOG_LEVEL_ERR, "Failed to parse address: '%s'", ip_str);
+        return NULL;
+    }
+
+    int step = 1;
+    char *last_dot = strrchr(ip_str, '.');
+    if (strcmp(last_dot + 1, "255") == 0)
+    {
```

Comment:
I guess inet_aton() should confirm that the IP has a dot in it except in the case that ip_str is just a number. Looks pretty good to me. I was thinking maybe a check for last_dot != NULL would be good but then thought about inet_aton() call above there.

From inet_aton(3) one of the forms the string can take:
```
       a         The  value  a  is  interpreted as a 32-bit value that is stored directly into the
                 binary address without any byte rearrangement.
```

Reply:
I had to replace this with ``inet_addr`` due to windows (in)compatibility. And that one seems to be stricter. However, this is an internal function and it's documentation says it has to be given an IP address with "*4 numbers separated by dots*". So I don't think this is a problem